integration-tests/chain-signatures/tests/cluster/mod.rs

```diff
@@ -0,0 +1,136 @@
+mod spawner;
+
+use std::collections::HashSet;
+
+use integration_tests_chain_signatures::local::NodeConfig;
+use mpc_contract::primitives::Participants;
+use near_workspaces::network::Sandbox;
+use spawner::ClusterSpawner;
+
+use mpc_contract::{ProtocolContractState, RunningContractState};
+use mpc_node::web::StateView;
+
+use anyhow::Context;
+use integration_tests_chain_signatures::containers::DockerClient;
+use integration_tests_chain_signatures::{utils, MultichainConfig, Nodes};
+use near_workspaces::{Account, AccountId, Contract, Worker};
+use url::Url;
+
+use crate::actions::sign::SignAction;
+use crate::actions::wait::WaitAction;
+
+pub fn spawn() -> ClusterSpawner {
+    ClusterSpawner {
+        wait_for_running: false,
+        cfg: MultichainConfig {
+            nodes: 3,
+            threshold: 2,
+            protocol: Default::default(),
+        },
+    }
+}
+
+pub struct Cluster {
+    pub cfg: MultichainConfig,
+    pub docker_client: DockerClient,
+    pub rpc_client: near_fetch::Client,
+    http_client: reqwest::Client,
+    pub(crate) nodes: Nodes,
+}
+
+impl Cluster {
+    pub fn len(&self) -> usize {
+        self.nodes.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.nodes.is_empty()
+    }
+
+    pub fn url(&self, id: usize) -> Url {
```

Comment:
Should we use the account id to identify the node?

Reply:
yeah more utilities can be added later, I'm just doing it based on what we have. We can have something like `url_by_id` eventually